CSS/style.css

```diff
@@ -0,0 +1,313 @@
+/* ------------- Tablet Media query Start Here --------- */
+
+@media screen and (min-width: 550px) {
+  .homepage-main {
+    grid-template-columns: 20vh auto auto auto;
+    grid-template-rows: 10rem 15rem 15rem 10rem;
+    grid-template-areas:
+      "hero     hero     hero     hero   "
+      "welcome  welcome  welcome  welcome"
+      "feature  feature  feature  feature"
+      "product  product  product  product"
+      "footer   footer   footer   footer ";
+  }
+
+  /* --------- Homepage Hero for Tablet ------------- */
+
+  .hamburger-menue {
+    padding-right: 3rem;
+  }
+
+  .hamburger-menue div {
+    width: 6vw;
+    height: 0.75vw;
+    margin: 0.5rem;
+  }
+
+  .hero-txt {
+    grid-column: 2 / span 3;
+    padding-right: 4rem;
+    padding-bottom: 1rem;
+    font-size: 2.5vw;
+  }
+
+  /* --------- Homepage Welcome for Tablet ------------- */
+
+  .homepage-welcome h1 {
+    font-size: 25px;
+  }
+
+  .homepage-welcome p {
+    font-size: 2vw;
+    line-height: 2;
+  }
+  /* --------- Homepage Feature for Tablet ------------- */
+
+  .homepage-feature {
+  }
+
+  /* --------- Homepage Product for Tablet ------------- */
+
+  .homepage-products {
+    display: flex;
+    flex-direction: row;
+    justify-content: space-around;
+    gap: 10px;
+  }
+
+  .homepage-products img {
+    width: auto;
+  }
+
+  /* --------- Footer for Tablet ------------- */
+
+  .footer-links {
+    display: grid;
+    grid-template-areas: "contact address phone email";
+    align-items: center;
+    padding-top: 0.75rem;
+  }
+
+  .footer-links div {
+    width: 20vw;
+    height: 1vw;
+    background-color: #6a6a6a;
+    margin: 0.4rem;
+  }
+}
+
+/* ------------- Desktop Media query Start Here --------- */
+/* ------------- Desktop Media query Start Here --------- */
+
+@media screen and (min-width: 900px) {
+  .homepage-main {
+    grid-template-columns: 5rem 1fr 1fr 1fr 1fr 5rem;
+    grid-template-rows: 17rem 3.5rem 15rem 8rem 10rem;
+    grid-template-areas:
+      "hero   hero     hero     hero     hero      hero"
+      ".      navbar   navbar   navbar   navbar       ."
+      ".      welcome  welcome  feature  feature        ."
+      ".      product  product  product  product      ."
+      "footer footer   footer   footer   footer  footer";
+  }
+
+  /* --------- Homepage Hero for Desktop ------------- */
+
+  .hamburger-menue {
+    display: none;
+  }
+
+  /* --------- Homepage Navbar for Desktop ------------- */
+
+  .navbar {
+    grid-column: navbar;
+    display: grid;
+    justify-items: end;
+    align-items: center;
+    background-color: antiquewhite;
+  }
+
+  .navbar-btn {
+    display: flex;
+    column-gap: 15px;
+    list-style: none;
+    padding-right: 20px;
+  }
+
+  .navbar-btn button {
+    background-color: orange;
+    border: none;
+    padding: 5px;
+    font-size: 12px;
+    font-weight: bold;
+  }
+
+  /* --------- Homepage Welcome for Desktop ------------- */
+
+  .homepage-welcome {
+    grid-column: welcome;
```

Comment:
for some screen size it overflows. You can solve this by:
either making it a scrollable div, if the design really requires that box to be of a certain size.
`overflow-y: scroll;`
or you could hack it doing a media query to make size smaller for these edge cases


or my favourite solution would be giving auto to the row so that it will adapt based on the content

<img width="489" alt="Screenshot 2022-11-12 at 23 12 57" src="https://user-images.githubusercontent.com/40305104/201496597-6903255a-d429-4976-84fd-270046c68ad7.png">